application/src/main/java/com/sanction/thunder/authentication/Key.java

```diff
@@ -0,0 +1,49 @@
+package com.sanction.thunder.authentication;
+
+import com.google.common.base.MoreObjects;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+public class Key {
+
+  private final String application;
+  private final String secret;
+
+  public Key(String application, String secret) {
```

Comment:
You're probably going to need a `@JsonCreator` annotation on this constructor. Also, I think you'll need `@JsonProperty()` annotations on the arguments. Check out the `StormUser.java` class to see what I mean.